README.md

```diff
@@ -2,727 +2,369 @@
+**Default:** `False`.
 
-* Note: This feature in implicitly ON.
-* Note: This feature can be turned off with command line switch ```--skip-retarget```.
+This controls whether a unique name is assigned to each platform. The unique name takes the form of `K64F[0]`, where the number between the brackets is an incrementing value. This name is accessible through the dictionary member `platform_unique_name` in the returned platform data. It defaults to `False`.
 
-## mbedls.json file properties
-* If file ```mbedls.json``` exists will be implicitly used to retarget results.
-* If file ```mbedls.json``` exists and flag ```--skip-retarget``` is set, there will be no retarget.
-* If file ```mbedls.json``` doesn't exist flag ```--skip-retarget``` has no effect.
+##### `read_details_txt`
 
-## Example of retargeting
-In this example we will replace serial port name during Freescale's K64F auto-detection:
-```
-$ mbedls
-+--------------+---------------------+------------+------------+-------------------------------------------------+
-|platform_name |platform_name_unique |mount_point |serial_port |target_id                                        |
-+--------------+---------------------+------------+------------+-------------------------------------------------+
-|K64F          |K64F[0]              |F:          |COM9        |0240022648cb1e77000000000000000000000000b512e3cf |
-+--------------+---------------------+------------+------------+-------------------------------------------------+
-```
+**Default:** `False`
 
-Our device is detected on port ```COM9``` and MSD is mounted on ```F:```. We can check more details using ```--json``` switch:
-```
-$ mbedls --json
-[
-    {
-        "mount_point": "F:",
-        "platform_name": "K64F",
-        "platform_name_unique": "K64F[0]",
-        "serial_port": "COM9",
-        "target_id": "0240022648cb1e77000000000000000000000000b512e3cf",
-        "target_id_mbed_htm": "0240022648cb1e77000000000000000000000000b512e3cf",
-        "target_id_usb_id": "0240022648cb1e77000000000000000000000000b512e3cf"
-    }
-]
-```
+This controls whether more data is pulled from the file system on each device. It can provide useful management data but also takes more time to execute.
 
-We must understand that ```mbed-ls``` stores information about mbed devices in dictionaries.
-The same information can be presented as dictionary where its keys are ```target_id``` and value is a mbed auto-detection data.
+---
 
-```
-$ mbedls --json-by-target-id
-{
-    "0240022648cb1e77000000000000000000000000b512e3cf": {
-        "mount_point": "F:",
-        "platform_name": "K64F",
-        "platform_name_unique": "K64F[0]",
-        "serial_port": "COM9",
-        "target_id": "0240022648cb1e77000000000000000000000000b512e3cf",
-        "target_id_mbed_htm": "0240022648cb1e77000000000000000000000000b512e3cf",
-        "target_id_usb_id": "0240022648cb1e77000000000000000000000000b512e3cf"
-    }
-}
+### `mock_manufacture_id(...)`
```

Comment:
Note to self: Prefix this function call with `mbeds.` just like `mbeds.list_mbeds()`.